### ISSUE TITLE
Sprint Nov 16

### ISSUE THREAD

**Issue by diasdavid** [ipfs/pm#55]
# Sprint Nov 16

## Sprint Goals

- 

## Sprint Discussions

[Schedule](https://github.com/ipfs/pm#sprint-discussion-schedule)

Please take notes in a separate pad, if you can, and link it here.

Endeavour         | Lead           | Pad    | Permanent Link
:---------------: | :------------: | :----: | :------------:
Apps on IPFS      | Juan Benet and Friedel Ziegelmayer | https://public.etherpad-mozilla.org/p/nov-16-apps-on-ipfs |  
infrastructure    | Lars Gierth    | https://public.etherpad-mozilla.org/p/nov-16-infrastructure |  
libp2p            | David Dias     | https://public.etherpad-mozilla.org/p/nov-16-libp2p |  
js-ipfs         | David Dias     | https://public.etherpad-mozilla.org/p/nov-16-js-ipfs |  
go-ipfs           | Jeromy Johnson | https://public.etherpad-mozilla.org/p/nov-16-go-ipfs | 
testing + ci      | Jeromy Johnson | https://public.etherpad-mozilla.org/p/nov-16-testing-ci |  
protocol + specs  | Juan Benet     | https://public.etherpad-mozilla.org/p/nov-16-protocol-specs |  
bitswap ml        | Juan Benet     | https://public.etherpad-mozilla.org/p/nov-16-bitswap-ml |  
Data Structures   | Juan Benet     | https://public.etherpad-mozilla.org/p/nov-16-data-structures |  

Please add the Agenda to the Pad before the endeavour sprint starts.

## Sprint Deliverables

- Add yourself below, that way we only have people listed who are really in the sprint
- Add links to issues down here. Only add things you can finish this sprint.

### [David Dias](@diasdavid)

- libp2p
  - [x] Test go-libp2p
  - [ ] Make libp2p-ping be a thing again (guarantees swarm compatibility)
  - [x] great sync with Jeromy on libp2p stuff while going through go-libp2p repo
- npm's registry on IPFS (on hold till 0.3.9 rebase on top of 0.4.0)
  - [x] install modules via IPFS using registry-mirror
  - [~] (ongoing) get Castor juiced up with all of the modules
  - [x] deploy new registry-mirror version on Castor
  - [x] Improve documentation and Roadmap across libp2p impl (providing some guidance to py-libp2p)
  - [x] drop deprecated impl of ipfs-blob-store https://github.com/ipfs/ipfs-blob-store/commit/409691cbbd6866a743247688f8e49e716e5f424e 
  - [x] upgrade ipfs-blob-store to latest js-ipfs-api https://github.com/ipfs/ipfs-blob-store/commit/bbc2e726161f9c76cb1ff692cfb18bdc16827d0a
- js-ipfs-api
  - [x] https://github.com/ipfs/js-ipfs-api/pull/131
  - [x] upgrade feat/files-api to use latest js-ipfs-api goodness (including speed boost by moving to Wreck)
  - [x] fix error handling https://github.com/ipfs/js-ipfs-api/commit/fc96f9305fd9f37b2a747becf2a11e5507089418
- js-ipfs
  - [ ] Continue working on jsipfs module https://github.com/ipfs/js-ipfs/pull/35
- extra
  - [x] Deliver Talk at NodeConf Barcelona
  - [x] Travel Logistics
  - [x] IPFS Lisbon Backshop (Thursday)
  - [ ] reinstall Go in my brain

### @lgierth

- go-ipfs
  - [x] fix gateway links with url rewrites ipfs/go-ipfs#1971 ipfs/infrastructure#116
  - [~] RFCR X-Ipfs-Gateway-Prefix hardening ipfs/go-ipfs#1988
  - [ ] document refs-denylists-dmca data structure
  - [~] RFCR fix seccat ipfs/go-ipfs#1987
- infrastructure
  - [~] open up cors on gateways (allow origin *) ipfs/infrastructure#109
  - [x] exclude dev0.4.0 hosts from pinbot ipfs/infrastructure#123
  - [ ] \<hostname>.h.ipfs.io pointing to cjdns addresses
  - [x] chat with gendale/alexandria
  - [x] nodejs 4 and npm for daviddias
- cjdns
  - [~] blocked by secio issue: discover peers from cjdns ipfs/go-ipfs#1316
  - [~] invite to fc00 development (aka go-cjdns) https://github.com/fc00/spec
  - [x] iptunnel setup with whyrusleeping
- hope
  - [ ] mfs api 500 responses
  - [ ] build a prometheus exporter using :5001
  - [ ] my todos for 0.4.0 migration

### [Richard Littauer](@RichardLitt)

- [ ] Stub out 5 different methods in HTTP API docs
- [ ] Sprint management
- [ ] Suggest community infrastructure for meetups
- [ ] Make the documentation around Commit Sign-Offs more friendly
- [ ] Solidify https://github.com/ipfs/community/issues/69

### @whyrusleeping

- [ ] ipfs-update
- [ ] add-mem fix
- [ ] ship 0.3.10
- [x] fs-repo-migrations for 0.4.0

### @VictorBjelkholm 

Main focus will be to move forward with PinCoop

* IPFS in JS
  * [ ] Help out to finish the API.md for js-ipfs-api
* PinCoop - https://github.com/victorbjelkholm/pincoop
  * [ ] having PinCoop ready for public consumption, mainly to scratch my own need for ipfsbin
* IPFSBin - https://github.com/victorbjelkholm/ipfsbin
  * [ ] start using PinCoop for ipfsbin, to make content more available
* Nodejs versions in IPFS
  * [ ] Make Castor to pin the already downloaded binaries
* Community
  * [ ] preparing talk for local meetups here in Barcelona 

### @Dignifiedquire 

- [ ] Bug fixing on js-ipfs-api
- [ ] Investigate next steps in webui refactoring (see above)

### @jbenet

- [x] QCon SF talk
- CR
  - [ ] ipfs-update
  - [ ] @rht's PRs
- spec work
  - [ ] fix repo spec for @masylum
- [x] sync with @kyledrake re permance + confs
- [ ] hacking archiving stuff with mek
- [x] various conversations
- [ ] org things

### @harlantwood

- [ ] plugging away at ipfs/ci-status#1
- [ ] considering options for ipfs/ci-status#6
- [ ] kicking off https://github.com/ipfs/infrastructure/issues/119

**Comment by kyledrake** [comment]
@kyledrake:

- [x] libp2p site
- [~] continue on DMCA
- [x] changes to blog post

I may be late for the sprint but will be joining the infrastructure hangout.

**Comment by Dignifiedquire** [comment]
- [~] Start mirroring npm-registry
- [~] Investigate and improve performance for `reginabox` and related modules https://github.com/davglass/follow-registry/pull/10
- [x] Refactor js-ipfsd-ctl and add ability to select a specific ipfs binary (needed for station packaging) https://github.com/ipfs/js-ipfsd-ctl/pull/33
- [x] Make ipfs-geoip more resilient https://github.com/ipfs/ipfs-geoip/pull/4
- [x] Use pretty print from ipfs-geoip in webui https://github.com/ipfs/webui/pull/93

**Comment by whyrusleeping** [comment]
## @whyrusleeping sprint update:
- [x] merge transports changes
- [x] rebase 0.4.0
- [x] extract go-libp2p
  - [ ] still working on vendoring everything nicely
- [x] gx modularity and hooks
  - [x] lots of gx fun
- [ ] merge add perf
  - [x] rebased on 0.4.0
- [x] go-datastore nosync flatfs
- [ ] addrsplosion part 4
- [x] dht query profiling
	- [x] kbucket nearest peers fix
	- [ ] dht query perf fixes (waiting to PR)
	- [ ] swarm dial backoff more aggressive (code complete)

**Comment by RichardLitt** [comment]
I did not have a good on-keyboard week this week, at all. I was AFK for most of it. I feel poorly about that. 

## IFPS HTTP docs
- [ ] Automatically populate a basic template as a one-off seed. Perhaps use ipfs/js-ipfs-api/pull/105.
- [x] Put a Swagger info bloc in ipfs/api.
- [ ] Look at httpie
- [ ] Document how to use curl for your own use to figure out what is missing in the API docs.
- [ ] Add a PR to ipfs.io/docs/api about how the commands there are not all-inclusive.
- [ ] Get `ipfs add` documented.
- [ ] Stub out the rest.
- [ ] Respond to ipfs/api/issues/1.

## READMEs
- [ ] Parse IPFS readme's automatically using a CLI tool for [ipfs-readme-standard](ipfs/ipfs-readme-standard).
- [ ] CR on @victorbjelkholm's ipfs/js-ipfs-api/pull/102
- [ ] Help with ipfs/ci-status/issues/1 and ipfs/infrastructure/issues/25.

## Community
- [x] Add ipfs/ci-status to PD.md. https://github.com/ipfs/ipfs/pull/123
- [x] Sprint management.
- [x] Go through and move ideas from ipfs/pm/issues/43 into ipfs/pm/README.md. https://github.com/ipfs/pm/pull/52
- [x] Edit ipfs/community/blob/master/docs/amending-commits.md. https://github.com/ipfs/community/pull/70
- [ ] Respond to issue about community infrastructure for meetups.
- [ ] Make the documentation around Commit Sign-Offs more friendly
- [ ] Template out issue reporting file from ipfs/go-ipfs/issues/1946.


## Added Done
- [x] Updated project-directory.md PR https://github.com/ipfs/ipfs/pull/116
- [x] Talked a lot about community entry pointes, added various tasks to my TODO. Cf https://github.com/ipfs/community/issues/69

**Comment by VictorBjelkholm** [comment]
Many things to do over at Typeform, got some time to spend on IPFS though...

### Update

**IPFS in JS**

- [X] Finish writing docs and API docs (Still haven't pushed the branch for this, will do tonight!)
- [ ] Help out to finish the API.md for js-ipfs-api
- [X] working on node-ipfs-api to pass the tests in browser + node env (didn't help very much with this...)

**PinCoop** - https://github.com/victorbjelkholm/pincoop

- [X] Rename to PinCoop
- [ ] having PinCoop ready for public consumption, mainly to scratch my own need for ipfsbin
- [X] Finish prototype with pull-model (did at the IPFS hackday)

**IPFSBin** - https://github.com/victorbjelkholm/ipfsbin

- [X] new ui for ipfsbin to be able to select language amongst other things
- [ ] start using PinCoop for ipfsbin, to make content more available

** nodejs version hosting on IPFS ** - https://github.com/victorbjelkholm/nodejs-mirror

- [X] Deploy
- [X] Download all versions
- [X] Publish
- [X] Write blog post about it https://gist.github.com/VictorBjelkholm/af38baab3728414ab478

**Community**

- [~] preparing talk for local meetups here in Barcelona 
- [X] collecting things for awesome-ipfs showcase
- [X] Barcelona hack day with IPFS and Redbooth

**Comment by lgierth** [comment]
### @lgierth's sprint update

- go-ipfs
  - [~] RFCR fix gateway links with url rewrites ipfs/go-ipfs#1971 ipfs/infrastructure#116
  - [x] deploy dev0.4.0 to castor and pluto ipfs/infrastructure#115
- infrastructure
  - [x] fix restarter ipfs/infrastructure@6dfef4b
  - [x] update ansible ipfs/infrastructure#114
  - [x] remove ipfs logging config, was thought to be broken, but actually removed! ipfs/go-ipfs#1959
  - [~] open up cors on gateways (allow origin *) ipfs/infrastructure#109
  - [ ] \<hostname>.h.ipfs.io pointing to cjdns addresses
- refs
  - [ ] dmca yyyy/mm/dd structure ipfs/refs#4
  - [x] writeup for kyledrake https://asciinema.org/a/7wgjaa5co0oo1lft5l1tvws43
- cjdns
  - [x] update to latest v17
  - [~] discover peers from cjdns ipfs/go-ipfs#1316
  - [~] invite to fc00 development (aka go-cjdns)
- org stuff
- hope
  - [ ] plan for community pinning / pincoop ipfs/infrastructure#113
  - [ ] plan for writable gateway ipfs/infrastructure#105

**Comment by ion1** [comment]
* [~] Slight progress with https://github.com/ipfs/go-ipfs/pull/1921, more work remains to be done.

**Comment by Dignifiedquire** [comment]
## Apps on IPFS

#### Lead: Friedel Ziegelmayer

## Agenda

### Components all the things
- https://github.com/ipfs/webui/pull/73
- https://github.com/ipfs/ipfs/issues/109
    
### Help and Report on station

## Participants

- Friedel Ziegelmayer @dignifiedquire
- David Dias @diasdavid 
- Johan Kiviniemi @ion1
- Victor @VictorBjelkholm 
- Juan Benet @jbenet 

## Notes

### Station

- Multiple daemons vs a single daemon
- TODO: Figure out a way to run multiple daemons with station
- TODO: Make a module to check for multicast announcements (using https://github.com/diasdavid/js-libp2p-mdns-discovery)
- TODO(Friedel): describe better what should happen here https://github.com/ipfs/station/issues/50 

### Components

- "Do not join the components war, just be the arms dealer"
- TODO: polish the webui & add tests!!!!!!
- TODO: Make the webui more modular
- TODO: Recipe for modular components on ipfs
- TODO: Use IPNS hash for the webui

Really great meeting :) Thanks everyone :)

**Comment by lgierth** [comment]
## Infrastructure

#### Lead: Lars Gierth

Hangout: https://hangouts.google.com/hangouts/_/ipfs.io/ipfs-infrastructure

## Agenda

- refs
  - go-ipfs side of refs-denylists-dmca
  - gateway prefix
- dev0.4.0
  - pinbot
  - repo migrations
- community pinning
  - ipfs-cluster
- writable gateway
  - put vs. post

## Participants

- ion
- jbenet briefly
- kyledrake
- lgierth
- VictorBjelkholm
- whyrusleeping

## Notes

- refs
  - why will take care of the go-ipfs side of refs-denylists-dmca (after 0.4.0)
  - lgierth will describe the changes to the dmca data structure
  - dmca: preliminary blocking using nginx
  - dmca link on legal page
- writable gateway
  - ipfs.io should probably be left readonly
  - saves us from hard problems like SEO spam, resource exhaustion
  - might be a space for businesses
- nodejs / iojs archives
  - lgierth & VictorBjelkholm coordinate regarding storage
- arin
  - kyledrake figures out registration and ovh support
  - lgierth figures out hetzner and digitalocean support

**Comment by diasdavid** [comment]
## libp2p

#### Lead: David Dias

## Agenda

- go-libp2p is out - goals for the week

**You have 30 minutes for this agenda**, 5 minutes before the meeting ends, consider moving the remaining items to a github discussion thread so that all the other sprint meetings can start in time.

## Participants

- David Dias
- Kyle Drake
- Juan Benet
- Friedel Ziegelmayer
- Johan Kiviniemi
- Victor Bjelkholm

## Notes

- Authentication in IPFS with key derivation (key) discussion
- https://github.com/ipfs/specs/tree/master/records
- https://github.com/ipfs/specs/tree/master/keychain
- https://github.com/ipfs/specs/issues/31
- https://github.com/ipfs/specs/issues/24

##### After sprint meeting is finished, create the respective action items on the Github sprint issue

**Comment by diasdavid** [comment]
## js-ipfs

#### Lead: David Dias

## Agenda



**You have 30 minutes for this agenda**, 5 minutes before the meeting ends, consider moving the remaining items to a github discussion thread so that all the other sprint meetings can start in time.

## Participants

- David Dias
- Kyle Drake
- Friedel Ziegelmayer
- Victor Bjelkholm

## Notes

- js-ipfs with delegated Node is an undergoing effort
- js-ipfs-repo is an undergoing endeavour, starter by @masylum
- js-ipfs on top of js-libp2p needs js-ipfs to be implemented first. 

##### After sprint meeting is finished, create the respective action items on the Github sprint issue

**Comment by whyrusleeping** [comment]
## @whyrusleeping TODO
- [ ] ipfs-update
- [ ] add-mem fix
- [ ] ship 0.3.10
- [ ] fs-repo-migrations for 0.4.0

**Comment by harlantwood** [comment]
- [ ] plugging away at ipfs/ci-status#1

**Comment by diasdavid** [comment]
Today I'll be flying at 17:55 for 2 hours and arrive at 18:55 in Lisbon (hacking the time!), I'll make as fast as I can home so I arrive on time for hangouts :)

**Comment by diasdavid** [comment]
### SPRINT CHECK IN

Going to drop my check in now as I'm about to board in a plane :)

- libp2p
  - [x] Test go-libp2p
  - [x] great sync with Jeromy on libp2p stuff while going through go-libp2p repo
- npm's registry on IPFS (on hold till 0.3.9 rebase on top of 0.4.0)
  - [x] install modules via IPFS using registry-mirror
  - [~] (ongoing) get Castor juiced up with all of the modules
  - [x] deploy new registry-mirror version on Castor
  - [x] Improve documentation and Roadmap across libp2p impl (providing some guidance to py-libp2p)
  - [x] drop deprecated impl of ipfs-blob-store https://github.com/ipfs/ipfs-blob-store/commit/409691cbbd6866a743247688f8e49e716e5f424e 
  - [x] upgrade ipfs-blob-store to latest js-ipfs-api https://github.com/ipfs/ipfs-blob-store/commit/bbc2e726161f9c76cb1ff692cfb18bdc16827d0a
- js-ipfs-api
  - [x] https://github.com/ipfs/js-ipfs-api/pull/131
  - [x] upgrade feat/files-api to use latest js-ipfs-api goodness (including speed boost by moving to Wreck)
  - [x] fix error handling https://github.com/ipfs/js-ipfs-api/commit/fc96f9305fd9f37b2a747becf2a11e5507089418
  - [x] Code Review over 9000 PRs from @Dignifiedquire :D :D 
- js-ipfs - Didn't have the chance to put time on this endeavour
- extra
  - [x] Deliver Talk at NodeConf Barcelona (Saturday)
  - [x] Travel Logistics
  - [x] IPFS Lisbon Backshop (Thursday)
  - [~] reinstall Go in my brain

**Comment by whyrusleeping** [comment]
## sprint update
- [ ] fs-repo-migrations (complete and working)
- [x] libp2p vendored
- [x] merge flatfs nosync
- [x] wire contexts into libp2p new-stream
- [ ] debugging close notify
- [x] gx work

**Comment by Dignifiedquire** [comment]
## Sprint Update

#### WebUI

- [x] Add new labels
- [x] Clean up issues list
- Start preparations for stats dashboard
  - [~] https://github.com/Dignifiedquire/js-ipfs-event-stream
  - [~] https://github.com/Dignifiedquire/ipfs-dash

#### js-ipfs-api

- [x] fix: Handle non json formatted errors (https://github.com/ipfs/js-ipfs-api/pull/120)
- [x] feat: Add for urls (https://github.com/ipfs/js-ipfs-api/pull/121)
- [x] refactor: Switch to [wreck](https://github.com/hapijs/wreck) and [webpack](http://webpack.github.io/) reducing the minified size by 50% (https://github.com/ipfs/js-ipfs-api/pull/125)
- [x] refactor: Consistent ES2015 usage and passing browser tests (https://github.com/ipfs/js-ipfs-api/pull/127)
- [x] fix: Correct header detection in the browser (https://github.com/ipfs/js-ipfs-api/pull/128)
- [x] feat: Add `diag/sys` command (https://github.com/ipfs/js-ipfs-api/pull/129)
- [x] feat: Add `log/tail` back for 0.3.9 (https://github.com/ipfs/js-ipfs-api/pull/130)
- [x] test: Split tests into seperate files (https://github.com/ipfs/js-ipfs-api/pull/134)
- [x] refactor: Split api definition into seperate files (https://github.com/ipfs/js-ipfs-api/pull/136)
- [x] chore: Add gulp release task (https://github.com/ipfs/js-ipfs-api/pull/137)
- [x] docs: Explore jsdoc@3  (https://github.com/ipfs/js-ipfs-api/pull/132)

#### diasdavid/registry-mirror

- [x] feat: Add host and port options (https://github.com/diasdavid/registry-mirror/pull/6)


#### HTTP API Spec

- [~] Look into apiary.io (http://docs.ipfs.apiary.io/)

**Comment by RichardLitt** [comment]
Update:

### Officially this week:

- [~] Stub out 5 different methods in HTTP API docs
- [x] Sprint management
- [ ] Suggest community infrastructure for meetups
- [ ] Make the documentation around Commit Sign-Offs more friendly
- [x] Solidify https://github.com/ipfs/community/issues/69
---

### Extra
- [x] PR awesome-ipfs saying it is only for finished projects. PR: ipfs/awesome-ipfs/pull/19.
- [x] PR ipfs/community adding a note that this is only for meta non-code discussions. PR: ipfs/community/pull/75.
- [x] PR ipfs/apps and ipfs/archives explaining in the README what they are for. Right now, those are vague. PR: https://github.com/ipfs/apps/commit/d2c522c12b20f7f195ceb0ecf1c913883470cca2. Issue: ipfs/archives/commit/cee8bb05dab4a56da1749c021aef56c427a0300f.
- [x] Make a note on ipfs/ipfs saying that the ipfs/ipfs FAQ is closed, open new ones in ipfs/faq. PR: ipfs/ipfs/pull/124.
- [x] Make a PR to IPFS/IPFS adding in ipfs.io/docs lists to good repos. PR: ipfs/ipfs/pull/124.
- [x] Rewrite ipfs/community/issues/69 and respond

**Comment by lgierth** [comment]
### @lgierth sprint update

- go-ipfs
  - [x] fix gateway links with url rewrites ipfs/go-ipfs#1971 ipfs/infrastructure#116
  - [~] RFCR X-Ipfs-Gateway-Prefix hardening ipfs/go-ipfs#1988
  - [ ] document refs-denylists-dmca data structure
  - [x] fix seccat ipfs/go-ipfs#1987
- infrastructure
  - [~] open up cors on gateways (allow origin *) ipfs/infrastructure#109
  - [x] exclude dev0.4.0 hosts from pinbot ipfs/infrastructure#123
  - [ ] \<hostname>.h.ipfs.io pointing to cjdns addresses
  - [x] chat with gendale/alexandria
  - [x] nodejs 4 and npm for daviddias
- cjdns
  - [~] blocked by nasty secio issue: discover peers from cjdns ipfs/go-ipfs#1316
  - [~] invite to fc00 development (aka go-cjdns) https://github.com/fc00/spec
  - [x] iptunnel setup with whyrusleeping
- hope
  - [ ] mfs api 500 responses
  - [ ] build a prometheus exporter using :5001
  - [ ] my todos for 0.4.0 migration

**Comment by jbenet** [comment]
### @jbenet

- [x] QCon SF talk
  - [x] talked with cmeik
- CR
  - [ ] ipfs-update -- still on it
  - [ ] @rht's PRs -- still on it
- spec work
  - [x] repo spec PR
  - [x] update ipld spec PR addressing CR
  - [x] update overviews spec PR
- [x] sync with @kyledrake re permance + confs
- [x] hacking archiving stuff with mek
  - [x] beginning of open journal efforts
  - [x] very productive discussion on POST
- [x] various conversations
- [x] org things
- [x] made https://github.com/jbenet/http2ipfs-web

**Comment by jbenet** [comment]
> hacking the time

https://www.youtube.com/watch?v=KEkrWRHCDQU

**Comment by Dignifiedquire** [comment]
@jbenet glorious way to end this spring :D

**Comment by masylum** [comment]
@masylum

[x] - Started js-repo implementation and reviewed specs
